framework/src/base/FEProblem.C

```diff
@@ -3459,10 +3459,14 @@ FEProblem::computeResidual(NonlinearImplicitSystem &/*sys*/, const NumericVector
   {
     computeResidualType(soln, residual, _kernel_type);
   }
-  catch(MooseException & e)
-  {
-    mooseError("Impossible");
```

Comment:
@jwpeterson - You have a stack trace from Andy's original example. In there he's throwing from a material but by my logic we should be catching from a material because they are computed downstream from a threaded loop of some kind. Do you still have that code branch? Why did that exception slip through? It really should have been handled already in one of the already existing empty catches:
https://github.com/idaholab/moose/blob/devel/framework/src/base/NonlinearSystem.C#L766-L771